src/systems/physics/Physics.cc

```diff
@@ -1700,6 +1705,57 @@ void PhysicsPrivate::CreateJointEntities(const EntityComponentManager &_ecm,
         {
           // No need to create this joint because it was already created when
           // parsing the model.
+
+          // Check if mimic constraint should be applied to this joint's axes.
+          using AxisIndex = std::size_t;
+          std::map<AxisIndex, sdf::JointAxis> jointAxisByIndex;
+          auto jointAxis = _ecm.Component<components::JointAxis>(_entity);
+          auto jointAxis2 = _ecm.Component<components::JointAxis2>(_entity);
+
+          if (jointAxis)
+          {
+            jointAxisByIndex[0] = jointAxis->Data();
+          }
+
+          if (jointAxis2)
+          {
+            jointAxisByIndex[1] = jointAxis2->Data();
+          }
+
+          for (auto &axisByIndex : jointAxisByIndex)
+          {
+            auto axisIndex = axisByIndex.first;
+            auto axis = axisByIndex.second;
+            if (auto mimic = axis.Mimic())
+            {
+              auto jointPtrMimic = this->entityJointMap
+                  .EntityCast<MimicConstraintJointFeatureList>(existingJoint);
```

Comment:
The joint needs to be added to `entityJointMap` first before calling `EntityCast`. This is done below (line 1759), so all this logic needs to go below that.

Reply:
thanks! fixed in https://github.com/gazebosim/gz-sim/pull/1838/commits/e1448b451d1c5f82e1ce8d0d1710a3e57eb00efd